api/src/DataPersister/ResetPasswordDataPersister.php

```diff
@@ -42,10 +45,15 @@ public function remove($data, array $context = []) {
     }
 
     public function beforeCreate(ResetPassword $data): ResetPassword {
+        $resp = $this->reCaptcha->verify($data->recaptchaToken);
+        if (!$resp->isSuccess()) {
+            throw new HttpException(422, 'ReCaptcha failed');
+        }
+
         $user = $this->userRepository->loadUserByIdentifier($data->email);
 
         if (null == $user) {
-            throw new Exception('Create request failed');
+            throw new HttpException(422, 'Create request failed');
```

Comment:
A bit late to the party: Mir ist gerade aufgefallen, dass wir hier einen Error schmeissen. 

Aus Security-Sicht ist das oft so implements, dass aus der Antwort zum Client nicht ersichtlich ist, ob der User existiert oder nicht. Nicht dringend, aber könnte man irgendwo auf eine Liste mit Improvements nehmen.